tempesta_fw/http.c

```diff
@@ -2420,6 +2423,20 @@ tfw_http_resp_cache(TfwHttpMsg *hmresp)
 	}
 
 	/*
+	 * FSM needs TfwHttpReq to record data. It needs TfwHttpResp as well.
+	 * It will be added to request later in tfw_http_resp_fwd(), but it's
+	 * needed now. Save previous (NULL) value for the sake of safety
+	*/
+	state = &hmresp->conn->state;
+	prev_state_obj = state->obj;
+	state->obj = req;
+	prev_resp = req->resp;
+	req->resp = hmresp;
+	tfw_gfsm_move(state, TFW_HTTP_FSM_RESP_MSG_FWD, NULL, 0);
+	state->obj = prev_state_obj;
+	req->resp = prev_resp;
```

Comment:
Yeah, bad stuff about existing GFSM code. I created #838 issue to extend GFSM in proper way.